hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -73,52 +70,61 @@ def __delete__(self, obj):
         raise AttributeError(u'Cannot delete TestCase')
 
 
-def find_breaking_runner(state_machine_factory, settings=None):
-    def is_breaking_run(runner):
-        try:
-            runner.run(state_machine_factory())
-            return False
-        except HypothesisException:
-            raise
-        except EXCEPTIONS_TO_FAIL:
-            verbose_report(traceback.format_exc)
-            return True
+def run_state_machine_as_test(state_machine_factory, settings=None):
+    """Run a state machine definition as a test, either silently doing nothing
+    or printing a minimal breaking program and raising an exception.
+
+    state_machine_factory is anything which returns an instance of
+    GenericStateMachine when called with no arguments - it can be a class or a
+    function. settings will be used to control the execution of the test.
+    """
     if settings is None:
         try:
             settings = state_machine_factory.TestCase.settings
+            assert isinstance(settings, Settings)
         except AttributeError:
             settings = Settings.default
 
-    search_strategy = StateMachineSearchStrategy(settings)
+    @settings
+    @given(st.data())
+    def run_state_machine(machine, data):
```

Comment:
Why are we passing `machine` in as an argument rather than just referring to it through the closure? (And why is it called `machine` when it's a factory for one?)

Reply:
So that failure reports identify which state machine failed - otherwise it's buried in the traceback.  (and I'll rename the argument)